zap/src/main/java/org/zaproxy/zap/extension/ascan/FilterPanel.java

```diff
@@ -0,0 +1,503 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascan;
+
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Vector;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import javax.swing.DefaultListModel;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.ListSelectionModel;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.SiteNode;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.extension.ascan.filters.GenericFilterUtility;
+import org.zaproxy.zap.extension.ascan.filters.ScanFilter;
+import org.zaproxy.zap.extension.ascan.filters.impl.HttpStatusCodeScanFilter;
+import org.zaproxy.zap.extension.ascan.filters.impl.MethodScanFilter;
+import org.zaproxy.zap.extension.ascan.filters.impl.TagScanFilter;
+import org.zaproxy.zap.extension.ascan.filters.impl.UrlPatternScanFilter;
+import org.zaproxy.zap.model.Target;
+import org.zaproxy.zap.utils.ZapLabel;
+import org.zaproxy.zap.view.LayoutHelper;
+
+/**
+ * FilterPanel is used as a Tab in {@link CustomScanDialog}. This tab gives various options for
+ * filtering the Active Scan messages based on the selection.
+ *
+ * @author KSASAN preetkaran20@gmail.com
+ * @since TODO add version
+ */
+public class FilterPanel extends JPanel {
```

Comment:
Lets make this package accessible instead, this is not for use by add-ons.

Reply:
ok.